1/initramfs/stack_pop.c

```diff
@@ -1,11 +1,15 @@
 #include <stdlib.h>
 #include "stack.h"
+#include <sys/stat.h>
+#include <fcntl.h>
 
 int main()
 {
-	while(1) {
-		printf("%d\n", stack_pop());
-		sleep(rand() % 5 + 1);
-	}
-	return 0;
+    while(1) {
+        printf("stack_pop: %d\n", stack_pop());
+        sleep(rand() % 5 + 1);
+        int fd = open("/sys/kernel/sys_stack/count", O_RDONLY);
```

Comment:
Ao abrir o arquivo, você recebe um file descriptor, que é um identificador desse arquivo. Cada processo tem uma tabela de arquivos abertos e os file descriptors começam em 0 e vão crescendo. Depois de abrir o arquivo, você precisa chamar [`read`](https://linux.die.net/man/2/read) nele pra ler o valor escrito no arquivo.

Para fazer uma nova leitura, ou você fecha e reabre o arquivo, ou chama [`lseek`](https://linux.die.net/man/2/lseek) pra voltar ao início dele.